control-plane/agents/core/src/core/reconciler/mod.rs

```diff
@@ -0,0 +1,49 @@
+mod persistent_store;
+pub mod poller;
+mod volume;
+
+use crate::core::task_poller::{PollContext, PollEvent, TaskPoller};
+use poller::ReconcilerWorker;
+
+use crate::core::registry::Registry;
+use parking_lot::Mutex;
+
+/// Used to start and stop the reconcile pollers
+#[derive(Debug)]
+pub(crate) struct ReconcilerControl {
+    worker: Mutex<Option<ReconcilerWorker>>,
+    event_channel: tokio::sync::mpsc::Sender<PollEvent>,
+    shutdown_channel: tokio::sync::mpsc::Sender<()>,
+}
+
+impl ReconcilerControl {
+    /// Return a new `Self`
+    pub(crate) fn new() -> Self {
+        let event_channel = tokio::sync::mpsc::channel(2);
```

Comment:
Oh there was no logic for 2 other than it's > than 1. I'm not sure what to set it to, I didn't want it too large to prevent queuing up loads of events. Any suggestion?